packages/microcosm/src/data.js

```diff
@@ -117,49 +132,52 @@ export function get(object: ?Object, keyPath: *, fallback?: *) {
 
 /**
  * Merge any number of objects into a provided object.
+ * $FlowFixMe This function enumerates on variable arguments
  */
-export function merge(...args: Array<?Object>): Object {
-  let copy = null
-  let subject = null
+export function merge(): Object {
+  let cloned = false
+  let copy = EMPTY_OBJECT
 
-  for (var i = 0, len = args.length; i < len; i++) {
-    if (isObject(args[i]) === false) {
+  for (var i = 0, len = arguments.length; i < len; i++) {
+    var next = arguments[i]
+
+    if (isObject(next) === false) {
       continue
     }
 
-    copy = copy || args[i] || {}
-
-    subject = subject || copy
+    if (copy === EMPTY_OBJECT) {
+      copy = next || EMPTY_OBJECT
+    }
 
-    var next = args[i]
     for (var key in next) {
       if (copy[key] !== next[key]) {
-        if (copy === subject) {
-          copy = clone(subject)
+        if (cloned === false) {
+          copy = clone(copy)
+          cloned = true
         }
 
         copy[key] = next[key]
       }
     }
   }
 
-  return copy || {}
+  return copy
 }
```

Comment:
Other update: `merge` is non-destructive. It returns a new object if any of the keys are different. Using a boolean flag to determine if we've already copied, and tweaking the guard on line 148, allowed me to get rid of the null state.

Reply:
any benefit to test this update?

Reply:
I had this before because I couldn't convince flow that I wasn't returning null, since `copy` was initialized as `null`.

However, thinking it more, we shouldn't ever return EMPTY_OBJECT because it is sealed and a user might try to add a key to it assuming it's "save".

So I should cover that case, for sure.